docs/concepts/norms.md

```diff
@@ -0,0 +1,37 @@
+# Norms
+
+Because rules are source code or shell commands, they don't come with a lot of
+inbuilt guardrails.  Rules are expected to behave themselves, which in a nutshell means:
+
+* Rules should generally be idempotent
+* Only modify (or create or delete) files under the working dir they're started in
+* Only read the files they declare as inputs (regardless of scope)
+* Only write the files they declare as outputs (regardless of scope)
+* Read whatever **external** (that is, outside the repo) state you want
+* Write **external** state only when the env var `YOLO` is set -- this disables
+  much of the parallelism in the name of getting the semantics right, because
+  we don't know if there's a way to undo.
+
+Ick promises that once a rule starts running, the working copy won't be further
+modified **by ick**.  You might have multiple simultaneous runs of the same
+scope=single-file rule in the same working copy, if there's reason to.
+
+For avoidance of doubt:
+
+You can read the git config of that working copy, or (for example) cache things
```

Comment:
This paragraph is word salad.  I'm trying to say that you **can** do A or B, but if you do A' or B' we don't recommend it (but we're a sign not a cop).